src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -140,6 +143,9 @@ void UpdateResources()
                     {
                         throw new Win32Exception(Marshal.GetLastWin32Error(), $"Could not set file permission {filePermissionOctal} for {appHostDestinationFilePath}.");
                     }
+
+                    if (enableMacOSCodeSign && RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                        CodeSign(appHostDestinationFilePath);
```

Comment:
Either here or in the SDK we should handle the case where we're on macOS but we're building for a different platform (Windows or Linux). In that case we should not attempt signing (as it will fail).

I "think" this should be done in the SDK since here we don't have a good way to detect the target platform (we kind of rely on the SDK to tell us the right things).